### ISSUE TITLE
High RAM & CPU utilization after 8h during long term run

### ISSUE THREAD

**Issue by erickubenka** [erickubenka/conviva#9]
**Describe the bug**
When using conviva for several hours, RAM will get flooded by somehwoe not garbage-collected objects. When the RAM is full, GC takes over all the availbale CPU und the bot becomes more and more unresponsible.

Things we already tried to tackle this issue:
* Switch to GC1 garbage collector
* Reduced permanent loop load when try to find new messages
* enabled verbose logging for GC

Possible problems that can probably lead to this error:
* Eithter the chrome and chromium session get more and more RAM usage, but it seems to be the java process in general (Check by adding a chrome refresh / restart during the execution - may with a simple command)
* object lists are not cleaned in a good way.
* underlying frameworks causing this issue

**Comment by erickubenka** [comment]
````
[36257.386s][info][gc] GC(37841) Pause Young (Normal) (G1 Evacuation Pause) (Evacuation Failure: Allocation) 2044M->2039M(2048M) 4.680ms
[36257.402s][info][gc] GC(37842) Pause Young (Concurrent Start) (G1 Evacuation Pause) (Evacuation Failure: Allocation) 2044M->2043M(2048M) 6.021ms
[36257.402s][info][gc] GC(37843) Concurrent Mark Cycle
[36257.412s][info][gc] GC(37844) Pause Young (Normal) (G1 Evacuation Pause) (Evacuation Failure: Allocation) 2044M->2044M(2048M) 5.350ms
[36261.299s][info][gc] GC(37845) Pause Full (G1 Compaction Pause) 2044M->1972M(2048M) 3886.089ms
[36261.299s][info][gc] GC(37843) Concurrent Mark Cycle 3897.238ms
05.05.2024 08:01:29.835 [main][WARN][SCID:20a35f8d81c11621aadc304224b6d956]: core.DesktopGuiElementCore - find() #277897 of UiElement(By.cssSelector: div.message-in div.copyable-text[32]) took 4.01s of 2s
[36261.542s][info][gc] GC(37846) Pause Young (Normal) (G1 Evacuation Pause) (Evacuation Failure: Allocation) 2044M->2039M(2048M) 5.132ms
[36261.563s][info][gc] GC(37847) Pause Young (Concurrent Start) (G1 Evacuation Pause) (Evacuation Failure: Allocation) 2044M->2044M(2048M) 6.490ms
[36261.563s][info][gc] GC(37849) Concurrent Mark Cycle
[36265.300s][info][gc] GC(37848) Pause Full (G1 Compaction Pause) 2044M->1972M(2048M) 3737.511ms
[36265.303s][info][gc] GC(37849) Concurrent Mark Cycle 3740.373ms
05.05.2024 08:01:33.831 [main][WARN][SCID:20a35f8d81c11621aadc304224b6d9
````

**Comment by erickubenka** [comment]
And Testerra is using the `EventFiringDecorator` and `WebDriverDecorator` of Selenium, which are currently flagged and annotaed as ` @Beta` class. 

Concrete the `DesktopWebDriverFactory` uses the `EventFiringDecorator` to decorate the driver in `setupNewWebDriverSession()`.

**Comment by erickubenka** [comment]
So basically, what I did to fix this issue:

Removed `io.testerra:driver-ui-desktop` and instead of that now uses `io.testerra.driver-ui` and implement the `DesktopWebDriverFactory` by our own. Therefor we used the `DesktopWebDriverFactory` class of Testerra itself and then removed the calls to the `@Beta` annotated classes for `EventFiringDecorator`.

This speeds up the whole element finding process and handling a lot. and now we can easily answer nearly in realtime. 
The memory leak was removed as well as ByteBuddy instances are not created anymore.
